routes/admin.js

```diff
@@ -20,49 +19,76 @@ router.get('/:adminId/pending-patients', async (req, res) => {
     }
 });
 
-router.post('/:adminId/approve-patient', async (req, res) => {
+router.get('/:adminId/pending-doctors', async (req, res) => {
     try {
-        const patientUid = req.body.userId;
         const adminId = new UserId(req.params.adminId);
         const adminRepository = new AdminRepository(req.app.locals.mongodb);
 
         const admin = new Administrator(adminId, adminRepository);
-        const response = await admin.approvePatient(patientUid);
+        const doctors = await admin.viewDoctors();
 
-        res.status(200).json({ data: response });
+        res.status(200).json({ doctors });
     } catch (error) {
         res.status(400).json({ error: error.message });
     }
 });
 
+router.get('/:adminId/pending-health-officer', async (req, res) => {
+    try {
+        const adminId = new UserId(req.params.adminId);
+        const adminRepository = new AdminRepository(req.app.locals.mongodb);
+
+        const admin = new Administrator(adminId, adminRepository);
+        const response = await admin.viewHealthOfficers();
 
-router.get('/:adminId/pending-doctors', async (req, res) => {
+        res.status(200).json({ data: response });
+    } catch (error) {
+        res.status(400).json({ error: error.message });
+    }
+});
+
+router.get('/:adminId/pending-immigration-officer', async (req, res) => {
     try {
         const adminId = new UserId(req.params.adminId);
         const adminRepository = new AdminRepository(req.app.locals.mongodb);
 
         const admin = new Administrator(adminId, adminRepository);
-        const doctors = await admin.viewDoctors();
+        const response = await admin.viewImmigrationOfficers();
 
-        res.status(200).json({ doctors });
+        res.status(200).json({ data: response });
     } catch (error) {
         res.status(400).json({ error: error.message });
     }
 });
 
-router.post('/:adminId/approve-doctor', async (req, res) => {
+router.post('/:adminId/approve-user', async (req, res) => {
     try {
-        const doctorUid = req.body.userId;
+        const userId = req.body.userId;
         const adminId = new UserId(req.params.adminId);
         const adminRepository = new AdminRepository(req.app.locals.mongodb);
 
         const admin = new Administrator(adminId, adminRepository);
-        const response = await admin.approveDoctor(doctorUid);
+        const response = await admin.approvePendingUser(userId);
 
         res.status(200).json({ data: response });
     } catch (error) {
         res.status(400).json({ error: error.message });
     }
 });
 
+router.post('/:adminId/reject-user', async (req, res) => {
+   try {
+       const userId = req.body.userId;
```

Comment:
Can a user be rejected after being approved? Right now if I reject a user, it rejects the user regardless if the user is pending or approved?